crypto-test/src/test/scala/org/bitcoins/crypto/ECPublicKeyTest.scala

```diff
@@ -93,20 +94,30 @@ class ECPublicKeyTest extends BitcoinSCryptoTest {
     val pubkey2 =
       ECPublicKey.fromBytes(ByteVector(firstByte) ++ pubkey1.bytes.tail)
 
-    val res1 = CryptoUtil.add(pubkey1, pubkey2)
-
-    assert(res1 == ECPublicKey.infinity)
+    assertThrows[AssertFailException](CryptoUtil.add(pubkey1, pubkey2))
 
     val decompressedPubkey1 =
       CryptoUtil.publicKeyConvert(pubkey1, compressed = false)
 
     val decompressedPubkey2 =
       CryptoUtil.publicKeyConvert(pubkey2, compressed = false)
 
-    val res2 =
-      CryptoUtil.add(decompressedPubkey1, decompressedPubkey2)
+    assertThrows[AssertFailException](
+      CryptoUtil.add(decompressedPubkey1, decompressedPubkey2))
+  }
 
-    assert(res2 == ECPublicKey.infinity)
+  it must "correctly compress keys" in {
+    forAll(CryptoGenerators.privateKey) { privKey =>
+      val pubKey = privKey.publicKey
+      val pubKeyCompressed = pubKey.compressed
+      val pubKeyDecompressed = pubKey.decompressed
+
+      assert(pubKey == pubKeyCompressed || pubKey == pubKeyDecompressed)
```

Comment:
Is there a reason we don't always return a compressed or decompressed key when calling `privKey.publicKey`? This should be deterministic

Reply:
it is determined by a param of `ECPrivateKey` currently